src/iohandler/process_io_handler.cc

```diff
@@ -240,26 +240,6 @@ size_t ProcessIOHandler::read(char* buf, size_t length)
         }
     }
 
-    if (num_bytes == 0) {
-        // not sure what we return here since no way of knowing about feof
-        // actually that will depend on the ret code of the process
-        ret = -1;
-
-        if (mainProc != nullptr) {
-            if (!mainProc->isAlive()) {
-                if (mainProc->getStatus() == EXIT_SUCCESS)
-                    ret = 0;
-
-            } else {
-                mainProc->kill();
-            }
-        } else
-            ret = 0;
-
-        killAll();
```

Comment:
Are you sure, we do not need this kind of cleanup. I am afraid of memory leaks or dead threads.

Reply:
I looked into it again. The transcoding process is always cleaned up eventually by calling `ProcessIOHandler::close()` either by the `BufferedIOHandler` class which holds the instance of `ProcessIOHandler`, or from the destructor. But with this modification it is only cleaned up when we transferred all content to the client, until then the transcoding process is in zombie state. For how much time it depends on the buffer settings, but likely for 10-20-30 seconds. I'll push changes to prevent this shortly.